lib/src/carry_save_multiplier.dart

```diff
@@ -4,29 +4,19 @@
 // carry_save_multiplier.dart
 // Implementation of pipeline multiplier module.
 //
-// 2023 February 21
-// Author: Max Korbel <max.korbel@intel.com>
+// 2023 May 19
+// Author: Yao Jing Quek <yao.jing.quek@intel.com>
 //
 
 import 'package:rohd/rohd.dart';
 
-/// A datatype class that hold result from [FullAdder].
-class FullAdderResult {
-  /// The output sum from full adder.
-  final sum = Logic(name: 'sum');
-
-  /// The output carry-out from full adder.
-  final cOut = Logic(name: 'c_out');
-}
-
-/// A [FullAdder] module that can perform simple adder. The full adder output
-/// can be defined in the truth table.
+/// A simple full-adder with inputs `a` and `b` to be added with a `carryIn`.
 class FullAdder extends Module {
-  /// The variable that hold [FullAdder] results.
-  final fullAdderResult = FullAdderResult();
+  /// The result [sum] from [FullAdder].
+  Logic get sum => output('sum');
 
-  /// The results returned from [FullAdder].
-  FullAdderResult get fullAdderRes => fullAdderResult;
+  /// The result [cOut] from [FullAdder].
```

Comment:
Clarify what this is, it's the carry bit.  Calling it `[cOut]` doesnt' help in the docs.